kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,257 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
     labeled_dataset = torch.utils.data.Subset(dataset, tindices)
     unlabeled_dataset = torch.utils.data.Subset(dataset, uindices)
     return labeled_dataset, unlabeled_dataset
+
+
+def domain_balanced_split(domain_labels, n_partitions, split_ratios):
+    """Get indices of random split. Samples with the same domain label will be split based on the given ratios.
+        Then the indices of different domains within the same split will be concatenated.
+
+    Args:
+        domain_labels (array-like): Labels to indicate the sample from which domain.
```

Comment:
English problem: *indicate which domain each sample is from* or simply "the domain label for each sample"